### PR TITLE
Depend on an unreleased version of AllenNLP

### PR DESCRIPTION
**Pull request by dirkgr**
The latest daily isn't out yet.

### CODE REVIEW

**Comment by epwalsh** [comment]
I'm confused by why we need this, and it will break the logic in `setup.py` https://github.com/allenai/allennlp-models/blob/master/setup.py#L30

**Comment by dirkgr** [comment]
Yeah, I know. I hope this fixes it. What a mess.

**Comment by dirkgr** [comment]
The root of the problem is that `requirements.txt` and `setup.py` need a different format for depending on URLs, but we treat them the same.

The fact that I wanted to do this for the allennlp dependency is somewhat orthogonal.

**Comment by dirkgr** [comment]
This changes the flag we use to override the allennlp dependency. I think it makes the code in `setup.py` more readable, but I don't know if it covers all the use cases we care about. @epwalsh?

**Comment by epwalsh** [comment]
I guess my concern is that in the CI for this repo, we wouldn't be testing against the latest commit of allennlp, and we'd have to keep manually updating the pinned commit.

**Comment by dirkgr** [comment]
Ah, yes. So we do want "exclude" behavior, not "override". But we also need a way to pin it to something other than the latest rc.

**Comment by epwalsh** [comment]
Having exclude behavior may not be necessary as long as there is a way to have the allennlp dependency completely unpinned.

**Comment by epwalsh** [comment]
Sorry for sneaking in a couple commits, just wanted to update the README before I forgot

File: README.md [hunk 1]
```diff
@@ -52,8 +52,10 @@ Once you have `allennlp` installed, run the following within the same Python env
 
 ```
 git clone https://github.com/allenai/allennlp-models.git
-EXCLUDE_ALLENNLP_IN_SETUP=true pip install -e .
+ALLENNLP_VERSION_OVERRIDE='allennlp' pip install -e .
 pip install -r dev-requirements.txt
 ```
 
+The `ALLENNLP_VERSION_OVERRIDE` flag ensures that your local install of `allennlp` won't be overridden. Otherwise `allennlp` will be intalled from the latest commit to the master branch on GitHub.
```
Comment (nelson-liu) [inline README.md:59]: s/intalled/installed/

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-allennlp
+git+https://github.com/allenai/allennlp.git
```
Comment (nelson-liu) [inline requirements.txt:1]: maybe explicitly specify `allennlp.git@master?`

File: README.md [hunk 3]
```diff
@@ -52,8 +52,10 @@ Once you have `allennlp` installed, run the following within the same Python env
 
 ```
 git clone https://github.com/allenai/allennlp-models.git
-EXCLUDE_ALLENNLP_IN_SETUP=true pip install -e .
+ALLENNLP_VERSION_OVERRIDE='allennlp' pip install -e .
```
Comment (nelson-liu) [inline README.md:55]: sorry this is an unrelated change, but you probably also want to add a `cd allennlp-models` above this line?

File: requirements.txt (hunk 2 above)
Comment (dirkgr) [reply requirements.txt:1]: sure

**Pull request merged**